### PR TITLE
Task/8 search area page

### PR DESCRIPTION
**Pull request by nonsensenv**
- #8  
- #9  
- #19 
## Problem
- No Search Area Page Established
- No Search Results Page Established
- No Database Connection (for testing purposes)  

> Added Search Results Page (Back-end Task) 

> Added User Can Search (Back-end Task)

> Added Temporary Database Connection [_Note: soon be changed by the official database server_]

> Added Nuxt Preset Files (e.g composables, api, use-case, services, etc.)

### CODE REVIEW

File: composables/use-paper.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export function usePaper() {
+  async function getResearchPaper() {
+    const response = await fetch("/api/get-research-paper"); // GET method by default
+    if (!response.ok) {
+      throw new Error("Failed to fetch research paper");
+    }
+    const data = await response.json();
+    const paper = data as Paper[];
+    return paper;
```
Comment (Techies-John) [inline composables/use-paper.ts:11]: add whitespace

File: components/search/SearchResults.vue [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+          class="img-poster"
+        />
+        <div class="img-title">{{ researchPaper.title }}</div>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script setup lang="ts">
+// import "../assets/bootstrap/bootstrap.min.css";
```
Comment (Techies-John) [inline components/search/SearchResults.vue:94]: remove if not used
```suggestion
// import "../assets/bootstrap/bootstrap.min.css";
```

File: components/search/SearchResults.vue [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+          placeholder="Search..."
+          autocomplete="off"
+          v-model="searchQuery"
+          @input=""
+          @focus="showSuggestions = true"
+        />
+
+        <!-- Search Suggestions Dropdown -->
+        <ul
+          v-if="showSuggestions && suggestions.length"
```
Comment (Techies-John) [inline components/search/SearchResults.vue:24]: no logics in template, use computed()
```suggestion
          v-if="showSuggestions && suggestions.length"
```

File: components/LoginPage.vue [hunk 4]
```diff
@@ -9,19 +9,40 @@
...
+                :type="passwordInputType"
+                id="password"
+                v-model="password"
+                class="form-control"
+                placeholder=""
+                required
+              />
+              <div
+                class="eye-icon"
+                v-if="password.length > 0"
```
Comment (Techies-John) [inline components/LoginPage.vue:35]: no logics in template, use computed()
```suggestion
                v-if="password.length > 0"
```

File: components/search/SearchArea.vue [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+          placeholder="Search..."
+          autocomplete="off"
+          v-model="searchQuery"
+          @input=""
+          @focus="showSuggestions = true"
+        />
+
+        <!-- Search Suggestions Dropdown -->
+        <ul
+          v-if="showSuggestions && suggestions.length"
```
Comment (Techies-John) [inline components/search/SearchArea.vue:23]: use computed
```suggestion
          v-if="showSuggestions && suggestions.length"
```

File: components/search/SearchArea.vue [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+import "../assets/global_style1/bootstrap.min.css";
+import { ref, onMounted, onBeforeUnmount, watch } from "vue";
+import debounce from "lodash/debounce";
+import { createClient } from "@supabase/supabase-js";
+
+// Initialize Supabase client
+const supabase = createClient(
+  "https://gfbbjwyqfnqpqhkwieyd.supabase.co",
+  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdmYmJqd3lxZm5xcHFoa3dpZXlkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MjY5NjM2OTMsImV4cCI6MjA0MjUzOTY5M30.5BACw-QIgGSV8HyfcKImuQoIrlAu9ez7w3J0Txf9s-8"
+);
```
Comment (Techies-John) [inline components/search/SearchArea.vue:77]: remove
```suggestion
// Initialize Supabase client
const supabase = createClient(
  "https://gfbbjwyqfnqpqhkwieyd.supabase.co",
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImdmYmJqd3lxZm5xcHFoa3dpZXlkIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MjY5NjM2OTMsImV4cCI6MjA0MjUzOTY5M30.5BACw-QIgGSV8HyfcKImuQoIrlAu9ez7w3J0Txf9s-8"
);
```

File: components/search/SearchArea.vue [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+      .sort((a, b) => b.score - a.score)
+      // Limit results to 5 suggestions
+      .slice(0, 5);
+
+    // Update suggestions with the ranked and sorted titles
+    suggestions.value = rankedSuggestions.map((s) => s.title);
+  }
+
+  shouldShowSuggestions.value = true;
+}, 300);
```
Comment (Techies-John) [inline components/search/SearchArea.vue:161]: No querying in ui layer, transfer it in services layer. Change the entire code.

File: components/search/SearchResults.vue [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+
+const { getResearchPaper } = usePaper();
+const research_papers = ref<Paper[]>([]);
+// Fetch data when the component is mounted
+onMounted(() => {
+  fetchPaper();
+});
+
+async function fetchPaper() {
+  console.log("hey man");
```
Comment (Techies-John) [inline components/search/SearchResults.vue:110]: remove console.logs

File: components/search/SearchResults.vue [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+      .sort((a, b) => b.score - a.score)
+      // Limit results to 5 suggestions
+      .slice(0, 5);
+
+    // Update suggestions with the ranked and sorted titles
+    suggestions.value = rankedSuggestions.map((s) => s.title);
+  }
+
+  shouldShowSuggestions.value = true;
+}, 300);
```
Comment (Techies-John) [inline components/search/SearchResults.vue:199]: No querying in ui layer, do it in services layer. Change the entire code.

File: server/api/get-research-paper/index.get.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { Paper } from "~/server/types/research-paper";
+import { getResearchPaperUseCase } from "~/server/use-case/get-paper";
+
+export default defineEventHandler(async (event) => {
+  // Call your use case to fetch the research papers
+  const data = await getResearchPaperUseCase();
+
+  console.log("Research papers fetched");
```
Comment (Techies-John) [inline server/api/get-research-paper/index.get.ts:8]: remove console.log

**Review by Techies-John — CHANGES_REQUESTED** [review]
change the following especially the query in ui layer

File: components/search/SearchResults.vue [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script setup lang="ts">
+// import "../assets/bootstrap/bootstrap.min.css";
+import { ref, onMounted, onBeforeUnmount, watch } from "vue";
+import debounce from "lodash/debounce";
+import { supabase } from "~/server/db/supabaseClient";
```
Comment (Techies-John) [inline components/search/SearchResults.vue:97]: remove
```suggestion
import { supabase } from "~/server/db/supabaseClient";
```

File: composables/use-paper.ts (hunk 1 above)
Comment (Techies-John) [inline composables/use-paper.ts:11]: add whitespace
```suggestion
    return paper;
```

**Review by Techies-John — CHANGES_REQUESTED** [review]

File: components/LoginPage.vue (hunk 4 above)
Comment (nonsensenv) [reply components/LoginPage.vue:35]: goods man

File: server/api/get-research-paper/index.get.ts (hunk 10 above)
Comment (nonsensenv) [reply server/api/get-research-paper/index.get.ts:8]: done

File: composables/use-paper.ts (hunk 1 above)
Comment (nonsensenv) [reply composables/use-paper.ts:11]: done

File: components/search/SearchResults.vue (hunk 8 above)
Comment (nonsensenv) [reply components/search/SearchResults.vue:110]: done

File: components/search/SearchResults.vue (hunk 2 above)
Comment (nonsensenv) [reply components/search/SearchResults.vue:94]: done

File: components/search/SearchArea.vue (hunk 5 above)
Comment (nonsensenv) [reply components/search/SearchArea.vue]: done

File: components/search/SearchArea.vue (hunk 6 above)
Comment (nonsensenv) [reply components/search/SearchArea.vue]: done

File: components/search/SearchArea.vue (hunk 7 above)
Comment (nonsensenv) [reply components/search/SearchArea.vue]: done

**Review by Techies-John — DISMISSED** [review]

**Pull request closed without merging**